File: my-folder/1894-merge-strings-alternately/solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
		char[] word1Arr = word1.toCharArray();
		char[] word2Arr = word2.toCharArray();
		int word1Len = word1Arr.length;
		int word2Len = word2Arr.length;
		char[]	result = new char[word1Len + word2Len];
		if (word1Len >= word2Len) {
			for (int i = 0; i < word2Len; i++) {
				result[i * 2] = word1Arr[i];
				result[i * 2 + 1] = word2Arr[i];
			}
			for (int i = word2Len * 2; i < word2Len + word1Len; i++) {
				result[i] = word1Arr[i-word2Len];
			}
		} else {
			for (int i = 0; i < word1Len; i++) {
				result[i * 2] = word1Arr[i];
				result[i * 2 + 1] = word2Arr[i];
			}
			for (int i = word1Len*2; i < word1Len + word2Len; i++) {
				result[i] = word2Arr[i-word1Len];
			}
		}
		return String.valueOf(result);
	}
}

File: my-folder/1894-merge-strings-alternately/solution.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int s1 = word1.size();
        int s2 = word2.size();

        string s;
        s.reserve(s1 + s2);  // Preallocate memory to avoid reallocation overhead

        int i = 0, j = 0;
        while (i < s1 || j < s2) {
            if (i < s1) s.push_back(word1[i++]);  // Use push_back instead of +=
            if (j < s2) s.push_back(word2[j++]);  
        }
        return s;
    }
};

